### PR TITLE
allow custom ulimits

### PR DESCRIPTION
**Pull request by joshgubler**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ customized solution you may need to use this code more as a pattern or guideline
 
 ```hcl
 module "my_app" {
-  source                       = "github.com/byu-oit/terraform-aws-fargate-api?ref=v3.5.0"
+  source                       = "github.com/byu-oit/terraform-aws-fargate-api?ref=v3.6.0"
```
Comment (GaryGSC) [inline README.md]:
```suggestion
  source                       = "github.com/byu-oit/terraform-aws-fargate-api?ref=v4.0.0"
```

File: examples/logging/logging.tf [hunk 2]
```diff
@@ -14,7 +14,7 @@ data "aws_elb_service_account" "main" {}
 //  name = "fake-example-cluster"
 //}
 module "fargate_api" {
-  source = "github.com/byu-oit/terraform-aws-fargate-api?ref=v3.5.0"
+  source = "github.com/byu-oit/terraform-aws-fargate-api?ref=v3.6.0"
```
Comment (GaryGSC) [inline examples/logging/logging.tf]:
```suggestion
  source = "github.com/byu-oit/terraform-aws-fargate-api?ref=v4.0.0"
```

File: examples/simple/simple.tf [hunk 3]
```diff
@@ -11,7 +11,7 @@ module "acs" {
 //  name = "fake-example-cluster"
 //}
 module "fargate_api" {
-  source = "github.com/byu-oit/terraform-aws-fargate-api?ref=v3.5.0"
+  source = "github.com/byu-oit/terraform-aws-fargate-api?ref=v3.6.0"
```
Comment (GaryGSC) [inline examples/simple/simple.tf]:
```suggestion
  source = "github.com/byu-oit/terraform-aws-fargate-api?ref=v4.0.0"
```

**Review by GaryGSC — CHANGES_REQUESTED** [review]
It's a shame there's still no convenient way to [make the new attribute optional](https://www.terraform.io/language/expressions/type-constraints#experimental-optional-object-type-attributes). Since everyone's going to be using `primary_container_definition`, this is breaking for every user.

Thanks for linking out to the docs.

This LGTM if we call it `v4.0.0`.

**Comment by joshgubler** [comment]
v4.0.0 it is

**Review by GaryGSC — APPROVED** [review]

**Pull request merged**